lang/src/org/partiql/lang/planner/transforms/AstToLogicalVisitorTransform.kt

```diff
@@ -0,0 +1,168 @@
+                is PartiqlAst.SetQuantifier.Distinct -> PartiqlLogical.build { call("filter_distinct", q) }
+            }
+        }
+
+    /**
+     * Throws [NotImplementedError] if any `SELECT` clauses were used that are not mappable to [PartiqlLogical].
+     *
+     * This function is temporary and will be removed when all the clauses of the `SELECT` expression are mappable
+     * to [PartiqlLogical].
+     */
+    private fun checkForUnsupportedSelectClauses(node: PartiqlAst.Expr.Select) {
+        when {
+            node.group != null -> TODO("Support for GROUP BY")
+            node.order != null -> TODO("Support for ORDER BY")
+            node.having != null -> TODO("Support for HAVING")
+        }
+    }
+
+    override fun transformLetBinding(node: PartiqlAst.LetBinding): PartiqlLogical.LetBinding =
+        PartiqlLogical.build {
+            letBinding(
+                transformExpr(node.expr),
+                varDecl_(node.name, node.name.metas),
+                node.metas
+            )
+        }
+
+    override fun transformStatementDml(node: PartiqlAst.Statement.Dml): PartiqlLogical.Statement {
+        TODO("Support for DML")
+    }
+
+    override fun transformStatementDdl(node: PartiqlAst.Statement.Ddl): PartiqlLogical.Statement {
+        TODO("Support for DDL")
+    }
+
+    override fun transformExprStruct(node: PartiqlAst.Expr.Struct): PartiqlLogical.Expr =
+        PartiqlLogical.build {
+            struct(
+                node.fields.map {
+                    structField(
+                        transformExpr(it.first),
+                        transformExpr(it.second)
+                    )
+                },
+                metas = node.metas
+            )
+        }
+}
+
+private object FromSourceToBexpr : PartiqlAst.FromSource.Converter<PartiqlLogical.Bexpr> {
+
+    override fun convertScan(node: PartiqlAst.FromSource.Scan): PartiqlLogical.Bexpr {
+        val asAlias = node.asAlias ?: errAstNotNormalized("Expected as alias to be non-null")
+        return PartiqlLogical.build {
+            scan(
+                AstToLogicalVisitorTransform.transformExpr(node.expr),
+                varDecl_(asAlias, asAlias.metas),
+                node.atAlias?.let { varDecl_(it, it.metas) },
+                node.byAlias?.let { varDecl_(it, it.metas) },
+                node.metas
+            )
+        }
+    }
+
+    override fun convertUnpivot(node: PartiqlAst.FromSource.Unpivot): PartiqlLogical.Bexpr {
+        TODO("Support for UNPIVOT")
+    }
+
+    override fun convertJoin(node: PartiqlAst.FromSource.Join): PartiqlLogical.Bexpr =
+        PartiqlLogical.build {
+            join(
+                joinType = AstToLogicalVisitorTransform.transformJoinType(node.type),
+                left = convert(node.left),
+                right = convert(node.right),
+                predicate = node.predicate?.let { AstToLogicalVisitorTransform.transformExpr(it) } ?: lit(ionBool(true)),
```

Comment:
Don't forget: feedback from #584 made `predicate` nullable to represent cross joins so you'll have to remove this the `.let { ... } ?: lit(...))` here....